### ISSUE TITLE
Unusual Behavior with Training Datasets Order and Dataset Size

### ISSUE THREAD

**Issue by Samir55** [Chat-3D/Chat-3D-v2#53]
Hi authors, 

Thank you very much for your great work! I have been experimenting with your model using the official training configuration provided, and I noticed some unusual behavior that I hope you can shed some light on.

## Observed Issue with Dataset Order
When training the model with the same datasets but changing the order of datasets in the train_tag (e.g., `multi3dref#scanrefer#obj_align#nr3d_caption#scan2cap` vs `scanrefer#obj_align#scan2cap#multi3dref#nr3d_caption`), I observed a significant difference in performance after the first epoch. Below are the results for reference:

#### Dataset Order: multi3dref#scanrefer#obj_align#nr3d_caption#scan2cap
```
"train_tag": "multi3dref#scanrefer#obj_align#nr3d_caption#scan2cap"
"val_tag": "scanrefer"

Start evaluating...
Evaluating scanrefer...
batch-size=32 length(#batches)=75
[epoch=0, global steps=1194] Val Results:
[scanrefer] Acc@0.25: 0.027146464646464648
[scanrefer] Acc@0.50: 0.02156986531986532
[scanrefer] Unique Acc@0.25: 0.03240477544059125
[scanrefer] Unique Acc@0.50: 0.023308698123934053
[scanrefer] Multiple Acc@0.25: 0.02595222724338283
[scanrefer] Multiple Acc@0.50: 0.02117495158166559
```

#### Dataset Order: scanrefer#obj_align#scan2cap#multi3dref#nr3d_caption
```
"train_tag": "scanrefer#obj_align#scan2cap#multi3dref#nr3d_caption"
"val_tag": "scanrefer"

Start evaluating...
Evaluating scanrefer...
batch-size=32 length(#batches)=75
[epoch=0, global steps=1194] Val Results:
[scanrefer] Acc@0.25: 0.4039351851851852
[scanrefer] Acc@0.50: 0.37110690235690236
[scanrefer] Unique Acc@0.25: 0.7288231949971575
[scanrefer] Unique Acc@0.50: 0.6759522455940875
[scanrefer] Multiple Acc@0.25: 0.3301484828921885
[scanrefer] Multiple Acc@0.50: 0.30187217559715945
```

These results suggest a drastic performance drop when datasets are listed in a different order, even though no dataset is removed. Could you provide any insights or explanations as to why this might happen?

## Question on Dataset Size and Model Performance

I have also noticed that the model's performance varies significantly depending on the size of the training dataset. For example, training the model only on ScanRefer results in poor performance, while combining datasets `(e.g., ScanRefer + Scan2Cap + Multi3DRef + Nr3DCaption + ObjAlign)` shows much better results. Below are results for training only on ScanRefer:

```
"train_tag": "scanrefer", 
"val_tag": "scanrefer",
Start evaluating...
Evaluating scanrefer...
batch-size=32 length(#batches)=75
[epoch=0, global steps=252] Val Results:
[scanrefer] Acc@0.25: 0.025147306397306397
[scanrefer] Acc@0.50: 0.01999158249158249
[scanrefer] Unique Acc@0.25: 0.025014212620807278
[scanrefer] Unique Acc@0.50: 0.018192154633314382
[scanrefer] Multiple Acc@0.25: 0.02517753389283408
[scanrefer] Multiple Acc@0.50: 0.02040025823111685
```

```
"train_tag": "scanrefer#obj_align", 
"val_tag": "scanrefer",
Start evaluating...
Evaluating scanrefer...
batch-size=32 length(#batches)=75
[epoch=0, global steps=434] Val Results:
[scanrefer] Acc@0.25: 0.02640993265993266
[scanrefer] Acc@0.50: 0.021464646464646464
[scanrefer] Unique Acc@0.25: 0.0244457077885162
[scanrefer] Unique Acc@0.50: 0.01648664013644116
[scanrefer] Multiple Acc@0.25: 0.02685603615235636
[scanrefer] Multiple Acc@0.50: 0.022595222724338282
```

Based on your experience, how large does the dataset need to be to ensure good model performance? Is there a minimum size or specific characteristics of a dataset that significantly impact model training?

I look forward to hearing your insights and recommendations. Thank you very much for your time and this valuable contribution to the community!

**Comment by ZzZZCHS** [comment]
Hi, thank you for your interest!

We’ve also noticed unusually poor performance in our experiments, especially on grounding tasks like ScanRefer and Multi3DRefer. As you mentioned, changing the dataset order (which is equivalent to shuffling the training data) could lead to poor results. This indicates that our current training process isn’t very stable.

We think this instability happens because the one-stage training method isn’t working well. The object ID tokens might not be trained properly (likely underfitting) when they are trained from scratch together with the 3D/2D projectors and LLM LoRA layers. Data quality may also contribute to the problem—current datasets may lack sufficient scale or diversity. Intuitively, constructing more comprehensive datasets could help the model better learn object ID tokens.

In our earlier version (Chat-3D v2.0), training was more stable because we didn’t use special tokens for object IDs. Instead, object IDs were represented as plain text (such as "Obj001"), which can be directly understood by the pretrained LLM. 

Therefore, potential solutions include finding a more robust method to learn these special tokens (e.g., more data or multi-stage training) or abandoning them altogether in favor of plain text representations for object IDs. We’re still working on this issue and hope to release a more stable version in the future.
